test/e2e/framework.go

```diff
@@ -939,6 +939,13 @@ func (data *TestData) waitForAntreaDaemonSetPods(timeout time.Duration) error {
 		currentNumAvailable := dsLinux.Status.NumberAvailable
 		UpdatedNumberScheduled := dsLinux.Status.UpdatedNumberScheduled
 
+		if clusterInfo.numNodes == 0 {
+			nodes, err := data.clientset.CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{})
+			if err != nil {
+				return false, err
+			}
+			clusterInfo.numNodes = len(nodes.Items)
```

Comment:
`clusterInfo` is a singleton while this is a multi-cluster env, I think it's not proper to set a value here, which would cause trouble when clusters have different number of nodes.
Instead, we should make the function not rely on `clusterInfo.numNodes`. It can get the desired numbers from the status of the DaemonSets (both antrea-agent and antrea-agent-windows). Note that it should only use DesiredNumberScheduled when `DaemonSet.Status.ObservedGeneration` equals to `DaemonSet.Generation` as the following comments explain.

Reply:
Thanks for the comment. I removed the assignment and use `UpdatedNumberScheduled` and `currentNumAvailable` to check if the DaemonSet is ready or not. In case, before we compare the above two variables, we need to check if `DaemonSet.Status.ObservedGeneration` equals to `DaemonSet.Generation`, both Linux DS and Windows DS.